### PR TITLE
Setup GitHub actions for build checks

### PR DESCRIPTION
**Pull request by jpenilla**
I noticed that the kotlin jvm target/toolchain version isn't set anywhere, that could be done to ensure compatibility with jdk 8+ no matter which jvm gradle is run with, however just making sure the published jar was built with 8 as has been done up until now is fine.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      fail-fast: true
+    steps:
+      - uses: actions/checkout@v2.3.5
+      - uses: gradle/wrapper-validation-action@v1
+      - name: JDK ${{ matrix.java }}
+        uses: actions/setup-java@v2.3.1
+        with:
+          java-version: ${{ matrix.java }}
+          cache: 'gradle'
+          distribution: 'temurin'
```
Comment (stephan-gh) [inline .github/workflows/build.yml:21]: Never heard of that, is that kind of a typical Java "distribution" people use nowadays? It's fine for me, just want to make sure this is nothing very special or something :)

**Review by stephan-gh — COMMENTED** [review]
This looks good to me, thanks a lot! Just one question:

File: .github/workflows/build.yml (hunk 1 above)
Comment (stephan-gh) [reply .github/workflows/build.yml:21]: Ah, looks like this is the replacement for AdoptOpenJDK. Clearly I'm not following Java development much anymore. :')

Comment (jpenilla) [reply .github/workflows/build.yml:21]: Yes, [Eclipse Temurin](https://projects.eclipse.org/proposals/eclipse-temurin) is the continuation of the [AdoptOpenJDK](https://adoptopenjdk.net/) project. [Amazon Coretto](https://aws.amazon.com/corretto/) and [Microsoft's OpenJDK builds](https://www.microsoft.com/openjdk) are two other widely used OpenJDK distributions, they all are pretty much the same as far as just building a project though. I tend to go with Temurin as in the past I've used AdoptOpenJDK, and on a good amount of linux distros installing the "openjdk" package will default to that as well, although again there isn't really much difference.

edit: github loaded in your second comment after I replied :smile:

**Comment by stephan-gh** [comment]
> I noticed that the kotlin jvm target/toolchain version isn't set anywhere, that could be done to ensure compatibility with jdk 8+ no matter which jvm gradle is run with, however just making sure the published jar was built with 8 as has been done up until now is fine.

FWIW: Apparently the Kotlin jvmTarget is set by the `kotlin-dsl` plugin to 1.8. There is still the slightly annoying warning in the log:

```
'compileJava' task (current target is 17) and 'compileKotlin' task (current target is 1.8) jvm target compatibility should be set to the same Java version.
```

But this seems to be a bug: see https://github.com/gradle/gradle issues/18935 (forgot the slash here on purpose to avoid a back link from the issue to this PR)

**Pull request merged**